e2e/README.md

```diff
@@ -37,3 +37,85 @@ If you have .env file, you can use CI or TEST_PROVIDER environment variables to
 - `TEST_PROVIDER=sauce npm run test:e2e`
 
 Both command do the same, use the saucelabs as a driver, but you get the results and the evidences locally.
+
+### pictures
+
+The results of the tests are written in the terminal, but you can check the created pictures about the elements in the same folder where the tests are located in the ./pics folder, next to the references.
+
+### clean the Tests
+
+If you have too many pics which is confusing you can run a command which clean every pics except of references.
+
+- `npm run clean-test`
+
+
+## Test developement guide
+
+The FEF tests based on visual testing. The basic concept is to take screenshot of every status of elements or pages and make difference to a reference picture.
+
+### Structure
+
+You can find everything regarding to end to end testing in the "e2e" folder. There are two important folders located in e2e: test and utils. The "test" contains the folders of the tests of the different pages. The "utils" contains the functions for the tests.
+
+### Test
+
+Every folder in the test folder contains a .js file and a pics folder. This is a whole page test, e.g. the [./e2e/test/select-tests](/e2e/test/select-tests/) folder contains every tests of [ux.Intellyo.com/select](http://ux.Intellyo.com/select).
+
+The tests located in the javascript file in groups.
+
+The references, screenshots and the diff pictures (created by the js file) located in the pics folder.
+
+The structure and the name of the pics are come from the structure of the test javascript file and the name of the test. For example the "this is the test set" describe contains a "I am the name of the test" test. This is stored in the "this-is-the-test-set" folder which contains a "I-am-the-name-of-the-test" folder (and other folders too). The pic of the element name generated from the last word of the test name, the time of the creation and the parameters of the the environment.
+
+ An example of a reference:
+  - e2e/test/buttons-tests/pics/reference/fef-buttons-tests/should-check-the-button:-yeah-button/yeah-button_reference_pic.png
+
+  An example of a screenshot:
+  - e2e/test/buttons-tests/pics/screen/fef-buttons-tests/should-check-the-button:-save-button/05-03-20-pm-2018-01-14_element_windows 10_element_chrome_v63_1400x982.png
+
+### Utils
+
+As it is mentioned above : The utils contains the functions for the tests. You can find here the tools for testing.
+
+The most important two function are the takeScreenshotAndGetWholePageCompareResult and takeScreenShotOfElement. The first takes picture of a whole page and compare to the reference. The second takes a screenshot of an element and make difference to the reference.
```

Comment:
1.  ~~The most  important two function~~ ➡️  The **two** most important function**s** are...

2. ~~and compare to the reference.~~ ➡️  ...and compare **it** to the reference.

3. ~~and make difference~~ ➡️  ...create a diff picture by comparing it to the reference picture.